### PR TITLE
Namespaces List should take a field selector

### PR DESCRIPTION
**Pull request by derekwaynecarr**
Aligns client with registry model in #5369 by allowing a field selector on list function.

### CODE REVIEW

File: plugin/pkg/admission/namespace/exists/admission.go [hunk 1]
```diff
@@ -85,7 +85,7 @@ func NewExists(c client.Interface) admission.Interface {
 	reflector := cache.NewReflector(
 		&cache.ListWatch{
 			ListFunc: func() (runtime.Object, error) {
-				return c.Namespaces().List(labels.Everything())
+				return c.Namespaces().List(labels.Everything(), labels.Everything())
```
Comment (bgrant0607) [inline plugin/pkg/admission/namespace/exists/admission.go]: Shouldn't the 2nd argument be fields (here and elsewhere)?

Comment (bgrant0607) [inline plugin/pkg/admission/namespace/exists/admission.go]: Or we don't have fields.Everything()?

Comment (derekwaynecarr) [inline plugin/pkg/admission/namespace/exists/admission.go]: We don't have a fields.Everything() yet.

Comment (derekwaynecarr) [inline plugin/pkg/admission/namespace/exists/admission.go]: sorry, i was mistaken.  we do have one now.  i will update.

Comment (derekwaynecarr) [inline plugin/pkg/admission/namespace/exists/admission.go]: It's a little confusing in the current state.  I am trying to find a current client that takes a fields.Selector as input.  Pod currently takes a label.Selector for both field and labels.  Is there a separate PR you are aware of that was looking to convert all of the instances of the old behavior ?

**Comment by derekwaynecarr** [comment]
It looks like the PR that moves to fields.Everything() is underway here:

https://github.com/GoogleCloudPlatform/kubernetes/pull/5233

Assuming that merges soon, I can make an update to this PR.

**Comment by bgrant0607** [comment]
@derekwaynecarr I'm ok with merging with a todo to switch to fields.Everything() later, if you'd like to do that.

File: plugin/pkg/admission/namespace/exists/admission.go (hunk 1 above)
Comment (bgrant0607) [inline plugin/pkg/admission/namespace/exists/admission.go]: #5171 introduced the fields pkg.

**Comment by derekwaynecarr** [comment]
If we can merge as is, I will follow-up with the change to uses the fields package.  I have some other work that needs to build on this.

**Pull request closed without merging**